crates/node/src/roles/cluster_controller.rs

```diff
@@ -61,11 +143,65 @@ impl ClusterControllerRole {
 }
 
 impl TryFrom<Options> for ClusterControllerRole {
-    type Error = Infallible;
+    type Error = ClusterControllerRoleBuildError;
 
     fn try_from(options: Options) -> Result<Self, Self::Error> {
+        let meta = options.meta.build(options.worker.kafka.clone())?;
+        let admin = options.admin.build(meta.schemas(), meta.meta_handle());
+
         Ok(ClusterControllerRole {
             controller: restate_cluster_controller::Service::new(options.cluster_controller),
+            admin,
+            meta,
         })
     }
 }
+
+#[derive(Debug, Clone)]
+struct GrpcWorkerHandle {
+    grpc_client: restate_node_services::worker::worker_client::WorkerClient<Channel>,
+}
+
+impl GrpcWorkerHandle {
+    fn new(channel: Channel) -> Self {
+        GrpcWorkerHandle {
+            grpc_client: restate_node_services::worker::worker_client::WorkerClient::new(channel),
+        }
+    }
+}
+
+impl Handle for GrpcWorkerHandle {
+    async fn terminate_invocation(
+        &self,
+        invocation_termination: InvocationTermination,
+    ) -> Result<(), Error> {
+        let invocation_termination =
+            bincode::serde::encode_to_vec(invocation_termination, bincode::config::standard())
+                .expect("serialization should work");
+
+        self.grpc_client
+            .clone()
```

Comment:
True. I first tried to go that way but then realized that it would affect all `restate_worker_api::Handle` implementations including the `WorkerCommandSender` which is used in a grpc service handler where I only get a `&self` passed in. That's why I went with this approach.